public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -50,39 +52,59 @@ function getNextRequestId() {
   return 'Q' + counter++;
 }
 
+export interface PanelConfig {
+  getTransformations: () => DataTransformerConfig[] | undefined;
+  getFieldOverrideOptions: () => ApplyFieldOverrideOptions | undefined;
```

Comment:
Not sure the name here is 100% as it's not the really the Panel configuration but DataConfiguration, and since it's not a data object I think it needs to be some kind of Source of some kind
* DataConfigSource
* DataConfigStore